screen_enumeration/getScreens.tentative.https.window.js

```diff
@@ -39,25 +39,27 @@ promise_test(async t => {
   await promise_rejects_dom(t, 'NotAllowedError', self.getScreens());
 }, 'getScreens() rejects the promise with permission denied');
 
-async_test(async t => {
+promise_test(async t => {
   await test_driver.set_permission({name: 'window-placement'}, 'granted');
   let iframe = document.body.appendChild(document.createElement('iframe'));
   assert_greater_than((await iframe.contentWindow.getScreens()).length, 0);
 
-  iframe.contentWindow.onunload = t.step_func(async () => {
-    // TODO(crbug.com/1106132): This should reject or resolve; not hang.
-    // assert_greater_than((await iframe.contentWindow.getScreens()).length, 0);
+  return new Promise(resolve => {
+    iframe.contentWindow.onunload = t.step_func(async () => {
```

Comment:
The style of this test ends up feeling like an odd mix still. `t.step_func(async () => { ... })` in general is suspicious because if any goes wrong after an `await` it won't be caught and results in a harness error.

I'd suggest making this function not async, calling `resolve()` right before the first await, then doing the rest afterwards. Whether you move the asserts out or not depend on whether the timing difference is important &ndash; run while the event is dispatching, or the first microtask following its dispatch.